vars/whitesourceExecuteScan.groovy

```diff
@@ -1,659 +1,21 @@
-                    dockerEnvVars: config.dockerEnvVars,
-                    dockerOptions: config.dockerOptions,
-                    dockerWorkspace: config.dockerWorkspace,
-                    stashContent: config.stashContent
-                ) {
-                    if (config.scanType == 'docker') {
-                        containerSaveImage script: parameters.script, containerImage: config.whitesource.scanImage, containerRegistryUrl: config.whitesource.scanImageRegistryUrl
-                    }
-                    if (config.whitesource.agentDownloadUrl) {
-                        def agentDownloadUrl = new GStringTemplateEngine().createTemplate(config.whitesource.agentDownloadUrl).make([config: config]).toString()
-                        //if agentDownloadUrl empty, rely on dockerImage to contain unifiedAgent correctly set up and available
-                        sh "curl ${script.env.HTTP_PROXY ? '--proxy ' + script.env.HTTP_PROXY + ' ' : ''}--location --output ${config.whitesource.agentFileName} ${agentDownloadUrl}".toString()
-                    }
-
-                    def javaCmd = 'java'
-                    if (config.whitesource.jreDownloadUrl) {
-                        //if jreDownloadUrl empty, rely on dockerImage to contain java correctly set up and available on the path
-                        sh "curl ${script.env.HTTP_PROXY ? '--proxy ' + script.env.HTTP_PROXY + ' ' : ''}--location --output jvm.tar.gz ${config.whitesource.jreDownloadUrl} && tar --strip-components=1 -xzf jvm.tar.gz".toString()
-                        javaCmd = './bin/java'
-                    }
-
-                    if(config.whitesource.installCommand)
-                        sh new GStringTemplateEngine().createTemplate(config.whitesource.installCommand).make([config: config]).toString()
-
-                    def options = ["-jar ${config.whitesource.agentFileName} -c \'${config.whitesource.configFilePath}\'"]
-                    if (config.whitesource.orgToken) options.push("-apiKey '${config.whitesource.orgToken}'")
-                    if (config.whitesource.userKey) options.push("-userKey '${config.whitesource.userKey}'")
-                    if (config.whitesource.productName) options.push("-product '${config.whitesource.productName}'")
-
-                    statusCode = sh(script: "${javaCmd} ${options.join(' ')} ${config.whitesource.agentParameters}", returnStatus: true)
-
-                    if (config.whitesource.agentDownloadUrl) {
-                        sh "rm -f ${config.whitesource.agentFileName}"
-                    }
-
-                    if (config.whitesource.jreDownloadUrl) {
-                        sh "rm -rf ./bin ./conf ./legal ./lib ./man"
-                        sh "rm -f jvm.tar.gz"
-                    }
+    final script = checkScript(this, parameters) ?: this
+    parameters = DownloadCacheUtils.injectDownloadCacheInParameters(script, parameters, BuildTool.MAVEN)
```

Comment:
Should be "MTA" to also inject npm package cache. (Doesn't really matter whether npm is used in the end.)

Reply:
Thanks! done ...